### PR TITLE
Refactor and comment resend related code

### LINKED ISSUE #192: Documentation and comments for resend related code

### PR DESCRIPTION
**Pull request by JekaMas**
closes #192

### CODE REVIEW

File: consensus/tendermint/backend/engine.go [hunk 1]
```diff
@@ -491,7 +491,7 @@ func (sb *Backend) Start(chain consensus.ChainReader, currentBlock func() *types
 	sb.coreStarted = true
 
 	sb.resend = make(chan messageToPeers, 1024)
-	go sb.ReSend(ctx, 10)
+	sb.ReSend(ctx, 10)
```
Comment (aneequesafdar) [inline consensus/tendermint/backend/engine.go]: Is it ok to block Start()?

Comment (JekaMas) [reply consensus/tendermint/backend/engine.go]: ok, it has been reverted :)

Comment (aneequesafdar) [reply consensus/tendermint/backend/engine.go]: Yeah looks good

**Comment by aneequesafdar** [comment]
LGTM, I am going to merge as @yazzaoui and @aesmonty can view these changes in `tendermint` branch.

**Pull request merged**